api/envoy/api/v2/listener/listener.proto

```diff
@@ -88,6 +89,28 @@ message FilterChainMatch {
   // [#not-implemented-hide:]
   google.protobuf.UInt32Value suffix_len = 5;
 
+  enum RequestSourceType {
+    // Any request origin matches
+    ANY_NETWORK = 0;
+    // Match a request that originates from the same host network or namespace
+    // This would only match if the connecting and listener IPs are the same.
+    // See EXTERNAL_NETWORK for an example.
+    SAME_NETWORK_NAMESPACE = 1;
```

Comment:
The naming is confusing, since it's clearly not the same network, but the same IP address.

Could we use `ANY`, `LOCAL` and `EXTERNAL` or is that too generic?

Reply:
@PiotrSikora I was following @rshriram 's comment:  https://github.com/envoyproxy/envoy/pull/4682#issuecomment-429307671
I don't have strong preference about the naming here.

Reply:
Why not `ANY`, `SAME`, `DISTINCT`? I don't think we gain much by pretending this is a general solution (see above discussion). What we're providing is an IP matching feature, it does not really imply local/external.

Reply:
Ok renamed the enum values to `ANY`, `SAME`, `DISTINCT`.

Reply:
@htuch I don't want to spend too much time bikeshedding, but it's not an IP matching feature (that's an implementation detail), it's about detecting whether the connection originated locally or externally.

What's misleading is the way this is implemented, i.e. checking if connecting client IP matches listening IP of the proxy, but the listening IP of the proxy doesn't have to be the destination IP, especially in scenarios where we intercept traffic using `iptables`, so this really is `local` vs `external`... That is, unless I'm misreading the discussion and reasoning behind this feature.